### PR TITLE
[TASK] Implement internal cache in PageProvider

### PR DESCRIPTION
**Pull request by bjo3rnf**
With reference to  #255

### CODE REVIEW

File: Classes/Provider/PageProvider.php [hunk 1]
```diff
@@ -298,22 +303,26 @@ protected function setDefaultValuesInFieldsWithInheritedValues(Form $form, array
...
-			$provider = $this->configurationService->resolvePrimaryConfigurationProvider($this->tableName, self::FIELD_NAME_SUB, $branch);
-			$form = $provider->getForm($branch);
-			if (NULL === $form) {
-				return $data;
-			}
-			$fields = $form->getFields();
-			$values = $provider->getFlexFormValuesSingle($branch);
-			foreach ($fields as $field) {
-				$values = $this->unsetInheritedValues($field, $values);
+		$cacheKey = $row['uid'];
```
Comment (NamelessCoder) [inline Classes/Provider/PageProvider.php]: Cache key should be at least table+field+uid to prevent collisions. Other than that, looks good!

**Comment by bjo3rnf** [comment]
The cache works as expected but the test ``PageProvider::canGetFlexformValuesUnderInheritanceConditions()`` fails now obviously. I am not sure what a sound test should look like now though.

**Comment by bjo3rnf** [comment]
The test fails because ``$cache`` is static. What now?

**Comment by NamelessCoder** [comment]
Huge thanks to @InvisibleKind for investigating and @bjo3rnf for making it work :)

**Comment by InvisibleKind** [comment]
You're welcome. Thanks for a fast solution and merge) That's what I love in Fluidtypo3 - quick resolution =)

**Comment by bjo3rnf** [comment]
Oh yes. That was one hell of a bug report, @InvisibleKind :+1:

**Pull request merged**